lib/task-queue/Page.js

```diff
@@ -0,0 +1,132 @@
+/* globals Task */
+"use strict";
+
+const {Storage} = require("lib/task-queue/Storage.js");
+const {Cu, Cc, Ci} = require("chrome");
+Cu.import("resource://gre/modules/Task.jsm");
+const {generateUUID} = Cc["@mozilla.org/uuid-generator;1"].getService(Ci.nsIUUIDGenerator);
+
+const DEFAULT_MAX_AGE = 1000 * 60 * 60 * 24 * 7; //one week
+
+class Page {
+  /**
+   * A page to be processed by the worker queue
+   *
+   * @param {String}  url
+   *                  the page URL
+   * @param {String}  [path]
+   *                  the path to the page DOM in the filesystem (Default: randomly generated)
+   * @param {Integer} [maxAge]
+   *                  The page max age (Default: DEFAULT_MAX_AGE)
+   * @param {Boolean} [remote]
+   *                  If the page was captured by navigating to it or by a remote process  (Default: false)
+   * @param {Integer} [createdAt]
+   *                  The page entry creation date (Default: now)
+   */
+  constructor({url, path, maxAge, remote, createdAt}) {
+    if (!url) {
+      throw (new TypeError("URL must be defined"));
+    }
+    this.url = url;
+    this.maxAge = maxAge || DEFAULT_MAX_AGE;
+    this.remote = Boolean(remote);
+    this.createdAt = createdAt || Date.now();
+    this.path = path || generateUUID().toString().slice(1, -1);
+    return Object.freeze(this);
```

Comment:
I did this because before the expiration date was calculated at creation time, so things would brake if Page wasn't immutable. I changed the implementation but forgot to change this accordingly.